### PR TITLE
Update dependencies, including newer Redux & React Router.

### PR DESCRIPTION
**Pull request by DFurnes**
### What does this PR do?

This pull request updates dependencies. In particular, I've Redux ([changelog](https://github.com/reduxjs/redux/releases/tag/v4.0.0)) & React Router ([changelog](https://github.com/ReactTraining/react-router/releases/tag/v5.0.0)) to their latest respective versions (in both cases, no changes were necessary). I've also removed [react-router-redux](https://github.com/reactjs/react-router-redux) because we were using an alpha version of a now abandoned project (and we weren't actually relying on any of its features).

### Any background context you want to provide?

I've clicked around and confirmed everything continues to work as expected, but as usual please let me know if you see anything spooky after pulling down and installing!

### What are the relevant tickets/cards?

N/A

### Checklist

* [ ] Documentation added for new features/changed endpoints.
* [ ] Added appropriate feature/unit tests.
* [ ] Added screenshot to PR description of related front-end updates on **small** screens.
* [ ] Added screenshot to PR description of related front-end updates on **medium** screens.
* [ ] Added screenshot to PR description of related front-end updates on **large** screens.

### CODE REVIEW

File: resources/assets/components/App.js [hunk 1]
```diff
@@ -31,15 +30,15 @@ const App = ({ store, history }) => {
         puckUrl={env('PUCK_URL')}
       >
         <ApolloProvider client={graphqlClient(env('GRAPHQL_URL'))}>
-          <ConnectedRouter history={history}>
```
Comment (mendelB) [inline resources/assets/components/App.js]: 👋

File: package.json [hunk 2]
```diff
@@ -38,6 +38,11 @@
       "@babel/preset-flow"
     ]
   },
+  "browserslist": [
```
Comment (mendelB) [inline package.json]: Where were we defining the legacy "browsers" previously?

Comment (DFurnes) [reply package.json]: Good Q! We were originally setting this [in our Webpack config](https://github.com/DoSomething/webpack-config/commit/fb08c9d50636c6414e2e4d5e232e3458d0d0a28a).

Comment (weerd) [reply package.json]: I think it does make more sense to set this as an option in the project instead of in the webpack config. I dig the change 💃

**Pull request merged**